pennylane/ops/identity.py

```diff
@@ -177,3 +176,159 @@ def adjoint(self):
 
     def pow(self, _):
         return [Identity(wires=self.wires)]
+
+
+class GlobalPhase(Operation):
+    r"""A global phase operation that multiplies all components of the state by :math:`e^{-i \phi}`.
+
+    **Details:**
+
+    * Number of wires: All (the operation acts on all wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        phi (TensorLike): the global phase
+        wires (Iterable[Any] or Any): unused argument - the operator is applied to all wires
```

Comment:
:thinking: I need to think about this, but are there cases where the wires matter? E.g., when there is control.

Reply:
It actually currently is used to determine the size of the matrix returned by `op.matrix`, so this is inaccurate now.

But beyond that, it seems to me that wires don't matter here. Which wire is the control wire matters, but that's not specified here, it's specified in `qml.ctrl(op, ctrl_wire)`. If the control wire is 1, then the global phase is applied to that whole term - the control wire isn't exempt. 

Maybe I'm not understanding this correctly, but as an example, what is the difference between passing `[0]`, `[1]` or `[0, 1]` as `wires` on the following:

```
dev = qml.device("default.qubit", 3)

@qml.qnode(dev)
def circuit():
    qml.Hadamard(2)
    qml.ctrl(qml.GlobalPhase(0.123, wires=[0]), 2)
    return qml.state()
```

Or conceptually even `[0, 1, 2]`, though in practice that isn't allowed by the validity checks on `qml.ctrl`. What is the significance of the `wires` argument (beyond defining `n_wires` for the matrix if people want to look at it)?

Reply:
The operator is applied without referring to the matrix:

```
def _apply_global_phase(self, state, operation: qml.GlobalPhase):  # pylint: disable=no-self-use
    """Applies a :class:`~.GlobalPhase` operation to the state."""
    return qml.math.exp(-1j * operation.data[0]) * state
```

Reply:
changed to `import Identity, GlobalPhase`

Reply:
discussed offline, decided this is fine for now